In [1]:
install.packages("readxl", repos = "http://cran.us.r-project.org")

Installing package into 'C:/Users/Der Shen Tan/Documents/R/win-library/3.3'
(as 'lib' is unspecified)


package 'readxl' successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\Der Shen Tan\AppData\Local\Temp\Rtmp0wOLCH\downloaded_packages


In [2]:
install.packages("tidyr", repos = "http://cran.us.r-project.org")

Installing package into 'C:/Users/Der Shen Tan/Documents/R/win-library/3.3'
(as 'lib' is unspecified)


package 'tidyr' successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\Der Shen Tan\AppData\Local\Temp\Rtmp0wOLCH\downloaded_packages


In [3]:
install.packages("scales", repos = "http://cran.us.r-project.org")

Installing package into 'C:/Users/Der Shen Tan/Documents/R/win-library/3.3'
(as 'lib' is unspecified)


package 'scales' successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\Der Shen Tan\AppData\Local\Temp\Rtmp0wOLCH\downloaded_packages


In [4]:
install.packages("ggrepel", repos = "http://cran.us.r-project.org")

Installing package into 'C:/Users/Der Shen Tan/Documents/R/win-library/3.3'
(as 'lib' is unspecified)


package 'ggrepel' successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\Der Shen Tan\AppData\Local\Temp\Rtmp0wOLCH\downloaded_packages


In [5]:
install.packages("dplyr", repos = "http://cran.us.r-project.org")

Installing package into 'C:/Users/Der Shen Tan/Documents/R/win-library/3.3'
(as 'lib' is unspecified)


package 'dplyr' successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\Der Shen Tan\AppData\Local\Temp\Rtmp0wOLCH\downloaded_packages


In [6]:
install.packages("class", repos = "http://cran.us.r-project.org")
library(class)

Installing package into 'C:/Users/Der Shen Tan/Documents/R/win-library/3.3'
(as 'lib' is unspecified)


package 'class' successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\Der Shen Tan\AppData\Local\Temp\Rtmp0wOLCH\downloaded_packages


Warning message:
"package 'class' was built under R version 3.3.3"

In [7]:
library(ggplot2)
library(tidyr)
library(dplyr)
library(scales)
library(ggrepel)
library(readxl)

Warning message:
"package 'ggplot2' was built under R version 3.3.3"Warning message:
"package 'tidyr' was built under R version 3.3.3"Warning message:
"package 'dplyr' was built under R version 3.3.3"
Attaching package: 'dplyr'

The following objects are masked from 'package:stats':

    filter, lag

The following objects are masked from 'package:base':

    intersect, setdiff, setequal, union

Warning message:
"package 'scales' was built under R version 3.3.3"Warning message:
"package 'ggrepel' was built under R version 3.3.3"Warning message:
"package 'readxl' was built under R version 3.3.3"

In [8]:
dat <- read.csv(file="June 10-July 12, 2015 - Gaming, Jobs and Broadband - CSV.csv", header=TRUE)

In [9]:
df <- as.data.frame(dat)

In [10]:
df

ï..psraid sample int_date lang usr cregion state form sex q1  ... ideo
1    100006    1      61015    1    S   3       51    1    2   1   ... 1   
2    100009    1      61015    1    U   3       24    2    2   2   ... 3   
3    100011    1      61015    1    S   2       39    2    2   1   ... 1   
4    100039    1      61015    1    S   3       37    2    2   1   ... 2   
5    100040    1      61015    1    U   3       21    1    2   1   ... 1   
6    100043    1      62215    1    R   2       20    1    2   1   ... 3   
7    100049    1      61015    1    R   1       42    1    1   1   ... 1   
8    100051    1      61915    1    R   2       26    1    1   1   ... 2   
9    100058    1      61015    1    R   3       40    2    1   2   ... 5   
10   100059    1      61015    1    U   1       23    1    1   3   ... 4   
11   100060    1      61015    1    R   2       20    1    2   1   ... 3   
12   100065    1      61915    1    R   3       40    1    2   3   ... 3   
13   100067    1      61015    1    R   2       27    1    1   1   ... 2   
14   100079    1      62315    1    U   3       12    1    2   3   ... 2   
15   100080    1      62015    1    S   3       45    2    2   4   ... 3   
16   100084    1      61015    1    R   3        1    2    2   2   ... 2   
17   100091    1      61915    1    S   3       21    2    1   1   ... 2   
18   100096    1      61015    1    R   3        5    2    1   3   ... 4   
19   100102    1      61015    1    U   1       36    2    2   1   ... 4   
20   100118    1      61015    1    U   3        1    2    1   1   ... 3   
21   100129    1      62315    1    R   4       30    1    2   4   ... 3   
22   100134    1      61015    1    S   2       20    1    1   2   ... 3   
23   100141    1      61015    1    R   3       48    2    2   2   ... 4   
24   100142    1      61015    1    S   3       45    2    1   4   ... 1   
25   100143    1      61015    1    U   1        9    2    1   1   ... 4   
26   100151    1      61015    1    U   3       51    1    2   2   ... 8   
27   100152    1      62415    1    S   4       49    1    2   2   ... 2   
28   100156    1      62115    1    S   2       27    2    2   1   ... 3   
29   100161    1      61015    1    S   3       12    2    2   2   ... 5   
30   100163    1      61015    1    S   2       27    2    2   2   ... 3   
...  ...       ...    ...      ...  ... ...     ...   ...  ... ...     ... 
1972 205355    2      71115    1    R   3       47    2    2   2   ... 1   
1973 205356    2      71115    1    R   2       18    2    2   1   ... 3   
1974 205363    2      71115    1    U   2       26    2    1   1   ... 4   
1975 205368    2      71115    1    R   4        4    2    1   3   ... 3   
1976 205371    2      71115    1    S   2       18    2    2   2   ... 4   
1977 205381    2      71115    1    S   2       29    1    2   1   ... 9   
1978 205390    2      71115    1    U   3       12    1    2   3   ... 4   
1979 205391    2      71115    1    S   3       13    1    1   1   ... 3   
1980 205392    2      71115    1    S   1       25    1    1   1   ... 3   
1981 205396    2      71115    1    U   3       47    2    2   2   ... 5   
1982 205398    2      71115    1    U   2       26    1    1   2   ... 2   
1983 205401    2      71115    1    S   2       26    2    2   2   ... 1   
1984 205410    2      71115    1    S   3        1    2    1   1   ... 2   
1985 205422    2      71115    1    U   4       53    2    2   1   ... 4   
1986 205425    2      71115    1    R   3       48    2    2   3   ... 2   
1987 205441    2      71115    1    S   3       40    2    1   1   ... 3   
1988 205448    2      71115    1    U   4        6    2    1   2   ... 3   
1989 205459    2      71215    1    S   2       29    1    2   2   ... 2   
1990 205460    2      71215    1    U   3       12    1    1   4   ... 2   
1991 205466    2      71215    1    R   4       35    1    1   1   ... 8   
1992 205469    2      71215    2    R   4        6   

In [12]:
names(df)

[1] "ï..psraid"  "sample"     "int_date"   "lang"       "usr"       
  [6] "cregion"    "state"      "form"       "sex"        "q1"        
 [11] "q3"         "eminuse"    "intmob"     "intfreq"    "home4nw"   
 [16] "bbhome1"    "bbhome2"    "bbhome3a"   "bbhome3b"   "bbhome3c"  
 [21] "bbhome3d"   "bbhome3e"   "device1a"   "smart1"     "q4"        
 [26] "bbsmart1"   "bbsmart2"   "bbsmart3a"  "bbsmart3b"  "bbsmart3c" 
 [31] "bbsmart3d"  "bbsmart3e"  "bbsmart3f"  "bbsmart4"   "cable1"    
 [36] "cable2"     "cable3a"    "cable3b"    "cable3c"    "act19"     
 [41] "act87a"     "act87b"     "act137"     "q5a"        "q5b"       
 [46] "q5c"        "q5d"        "q5e"        "q5f"        "date1a"    
 [51] "date2a"     "date3a"     "date3b"     "date4a"     "date4b"    
 [56] "date4c"     "date4d"     "date4e"     "date4f"     "game1"     
 [61] "game2a"     "game2b"     "game2c"     "game2d"     "game2e"    
 [66] "game2f"     "game3a"     "game3b"     "game4"      "emplnw"    
 [71] "emptype1"   "emptype2"   "emptype3"   "stud"       "job1a"     
 [76] "job1b"      "job1c"      "job1d"      "job1e"      "job1f"     
 [81] "job2"       "job3a"      "job3b"      "job3c"      "job3d"     
 [86] "job3e"      "job3f"      "job3g"      "job3h"      "job4"      
 [91] "smjob1"     "smjob2a"    "smjob2b"    "smjob2c"    "smjob2d"   
 [96] "smjob2e"    "smjob3"     "smjob4a"    "smjob4b"    "smjob4c"   
[101] "smjob4d"    "smjob4e"    "snsjob1a"   "snsjob1b"   "snsjob1c"  
[106] "snsjob2a"   "snsjob2b"   "job5a"      "job5b"      "job5c"     
[111] "job5d"      "auto1a"     "auto1b"     "auto1c"     "auto1d"    
[116] "auto1e"     "auto2"      "auto3"      "age"        "marital"   
[121] "hh1"        "par"        "kidage1"    "kidage2"    "kidage3"   
[126] "educ2"      "edinst"     "disa"       "party"      "partyln"   
[131] "ideo"       "hisp"       "birth_hisp" "race"       "inc"       
[136] "ql1a"       "qc1"        "zipcode"    "weight"     "standwt"

In [22]:
myvars <- c("age", "sex", "intfreq", "game4","emplnw")

In [23]:
newdf <- df[myvars]

In [24]:
newdf <- na.omit(newdf)

In [27]:
ndf <- newdf[ which(newdf$sex < 3 & newdf$intfreq < 6 & newdf$game4 < 3 & newdf$emplnw < 5), ]


In [28]:
ndf

age sex intfreq game4 emplnw
2    63  2   5       2     3     
5    65  2   3       2     3     
12   77  2   2       2     3     
13   52  1   1       1     1     
14   74  2   2       2     3     
15   43  2   2       2     1     
21   31  2   1       2     1     
27   39  2   2       2     4     
31   66  2   1       2     3     
33   19  1   3       2     2     
34   28  1   5       2     2     
36   45  2   2       2     1     
39   52  2   2       1     3     
42   80  2   4       2     3     
44   19  2   3       2     2     
46   25  1   2       1     1     
48   75  2   3       2     3     
50   73  2   2       2     1     
60   43  2   2       2     1     
69   66  2   3       1     3     
70   67  2   1       2     3     
84   51  2   2       2     1     
85   58  1   2       1     3     
87   34  1   2       1     1     
90   73  1   3       2     3     
91   51  2   2       2     2     
94   36  2   2       2     2     
97   77  1   2       2     3     
98   82  2   1       2     3     
99   73  2   4       2     3     
...  ... ... ...     ...   ...   
1926 26  1   1       2     2     
1927 53  2   2       2     1     
1929 73  2   4       2     3     
1932 49  1   2       2     1     
1934 37  2   1       2     1     
1940 46  2   2       2     1     
1941 20  1   1       2     4     
1943 26  2   1       1     1     
1945 22  2   2       2     2     
1946 43  2   2       2     1     
1948 64  1   4       2     3     
1949 56  2   3       2     1     
1951 23  1   1       2     2     
1957 22  1   5       2     4     
1958 50  2   2       2     1     
1960 25  1   1       2     1     
1961 35  1   2       2     1     
1963 26  2   2       1     4     
1966 33  1   2       2     1     
1970 28  2   2       2     1     
1971 23  1   3       1     4     
1976 29  2   2       2     4     
1978 35  2   1       1     1     
1979 26  1   2       1     2     
1982 27  1   2       2     4     
1986 36  2   1       2     2     
1987 62  1   1       2     1     
1996 35  1   1       2     1     
1998 77  2   5       2     3     
1999 40  2   2       2     2

How does the value in the column corresponds to their answer:

Age: Their age of the time they did the survey

Sex:
1) Male
2) Female

Internet uses frequency, Intfreq: 
1) Almost constantly
2) Several times a day
3) About once a day
4) Several times a week
5) Less often?

Gamer?, game4:
1) Yes, gamer
2) No, not gamer

Are they employed, emplnw:
1) Employed full-time
2) Employed part-time
3) Retired
4) Not employed for pay

5 Cluster

In [29]:
set.seed(76964057) #Set the seed for reproducibility
k <-kmeans(ndf, centers=5) #Create 5 clusters, Remove columns 1 and 2
k$centers #Display&nbsp;cluster centers
table(k$cluster) #Give a count of data points in each cluster

age      sex      intfreq  game4    emplnw  
1 47.69945 1.530055 2.169399 1.890710 1.688525
2 34.11798 1.426966 1.769663 1.758427 1.601124
3 61.99533 1.635514 2.369159 1.906542 2.242991
4 22.83069 1.333333 1.751323 1.666667 2.031746
5 81.06349 1.666667 2.714286 1.936508 2.698413


  1   2   3   4   5 
183 178 214 189  63 

Cluster 1) There 183 members in this group. They are mostly at the age of around 48 and consist of equally male and female. This group also uses internet several times a day and are mostly not gamers, which most likely that they are either employed full-time or part-time.


Cluster 2) There 178 members in this group. THey are mostly at the age of around 34 and consist of slightly more male partcitipant. This group also uses internet quite constantly(greater than cluster 1) maybe more times a day. They are also mostly not gamers but with a higher percent of gamer than cluster 1 and are also mostly employed, either full time or part-time.


Cluster 3) There are 214 members in this group . They are mostly at the age of 62 and consist of a bit larger female participant. This group uses internet about severals times a day with a chance to use it once a day. They are also mostly non gamers and are still employed part-time or retired. I believe that it could be there a high chance that it has high amount of retired members that could have skewed the data toward being employed part time.

Cluster 4) There are 189 members in this group and they are mostly at the age of 23. They are mostly male in this group and have a highest usage of internet in all the clusters, which could means heavy usage of internet to almost constantly. They also have the lagest amount of gamers compared to the other cluster and are mostly employed part time.


Cluster 5) There are 63 members in this group and they are mostly at the age of 81, once could assume smaller membership in this group because of old age. This group consist of the highest female participant, that barely use internet to almost once a day. They are almost all not gamers, and are the cluster with the least gamers. They are also mostly retired with a few exception of being employed part time.

3 Cluster

In [30]:
set.seed(76964058) #Set the seed for reproducibility
k <-kmeans(ndf, centers=3) #Create 5 clusters, Remove columns 1 and 2
k$centers #Display&nbsp;cluster centers
table(k$cluster) #Give a count of data points in each cluster

age      sex      intfreq  game4    emplnw  
1 72.32484 1.687898 2.573248 1.923567 2.687898
2 28.30518 1.378747 1.760218 1.711172 1.822888
3 51.97360 1.551155 2.214521 1.894389 1.772277


  1   2   3 
157 367 303 

Cluster 1) They have a group membership of about 157, with the age around 72 and are have greater female participant. This group also have their internet usage at about once a day to maybe a bit more than once a day. They are a practically non gamers with a few outlier and are mostly retired or employed part time.


Cluster 2) They have the largest group membership of about 367 members, with the average age of 28. This group have the highest male participant and have the greatest internet compared to other clusters, which is about several times a day to almost constantly. THey also have the greatest number of gamers and are mostly either working part-time or being employed full- time.

Cluster 3) They have quite a big membership in this group too, about 303 members. They are ages around 51 and have an equal amount of male and female participant. They uses internet about several times a day and have a good sum of gamers in their members. They are either mostly employed full-time or working part time. I think this group have the largest work force but may be skewed as it close to the age of retirement and which upset the data.